External/FEXCore/Source/Interface/Core/OpcodeDispatcher.cpp

```diff
@@ -1165,14 +1165,9 @@ void OpDispatchBuilder::SHLDOp(OpcodeArgs) {
   auto ShiftRight = _Sub(_Constant(Size), Shift);
 
   OrderedNode *Res{};
-  if (Size == 16) {
-    LogMan::Msg::A("Unsupported Op");
-  }
-  else {
-    auto Tmp1 = _Lshl(Dest, Shift);
-    auto Tmp2 = _Lshr(Src, ShiftRight);
-    Res = _Or(Tmp1, Tmp2);
-  }
+  auto Tmp1 = _Lshl(Dest, Shift);
```

Comment:
does it take care of Size here automagically?

Reply:
Yes, the sizing magic comes from the ops operating at the correct sizes (32bit/64bit) and the sizes being baked in to the instruction encodings that the frontend parse for us.